libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/components/avatar/AvatarData.kt

```diff
@@ -27,24 +28,28 @@ data class AvatarData(
                     startIndex++
                 }
 
-                var length = 1
-                var first = dn[startIndex]
+                var next = dn[startIndex]
 
                 // LEFT-TO-RIGHT MARK
-                if (dn.length >= 2 && 0x200e == first.code) {
+                if (dn.length >= 2 && 0x200e == next.code) {
                     startIndex++
-                    first = dn[startIndex]
+                    next = dn[startIndex]
                 }
 
-                // check if it’s the start of a surrogate pair
-                if (first.code in 0xD800..0xDBFF && dn.length > startIndex + 1) {
-                    val second = dn[startIndex + 1]
-                    if (second.code in 0xDC00..0xDFFF) {
-                        length++
+                while (next.isWhitespace()) {
+                    if (dn.length > startIndex + 1) {
+                        startIndex++
+                        next = dn[startIndex]
+                    } else {
+                        break
                     }
                 }
 
-                dn.substring(startIndex, startIndex + length)
+                val fullCharacterIterator = BreakIterator.getCharacterInstance()
```

Comment:
This will return a `BreakIterator` for the default locale, I am wondering what will happen if dn contains chars for another locale.

Reply:
I'm not sure if we should have different results for different locales, to be honest, it would be weird if an avatar looked different in devices with different locales 🫤 .